src/core/public/application/application_service.tsx

```diff
@@ -105,23 +130,41 @@ export class ApplicationService {
     }
 
     // Disable registration of new applications
-    this.apps$.complete();
-    this.legacyApps$.complete();
+    this.started = true;
 
     const legacyMode = injectedMetadata.getLegacyMode();
     const currentAppId$ = new BehaviorSubject<string | undefined>(undefined);
-    const { availableApps, availableLegacyApps, capabilities } = await this.capabilities.start({
-      apps: new Map([...this.apps$.value].map(([id, { app }]) => [id, app])),
-      legacyApps: this.legacyApps$.value,
+    const { availableApps, capabilities } = await this.capabilities.start({
+      apps: new Map(
+        [...this.apps].map(([id, appBox]) => [
+          id,
+          (isLegacyApp(appBox) ? appBox : appBox.app) as App | LegacyApp,
+        ])
+      ),
       injectedMetadata,
     });
 
     // Only setup history if we're not in legacy mode
     const history = legacyMode ? null : createBrowserHistory({ basename: http.basePath.get() });
 
+    const availableApps$ = new BehaviorSubject<ReadonlyMap<string, App | LegacyApp>>(availableApps);
+
+    combineLatest([
+      of(availableApps),
+      this.statusUpdaters$.pipe(map(statusUpdaters => [...statusUpdaters.values()])),
+    ])
+      .pipe(
+        map(([apps, statusUpdaters]) => {
+          return new Map([...apps].map(([id, app]) => [id, updateStatus(app, statusUpdaters)]));
```

Comment:
from https://github.com/elastic/kibana/issues/45291#issuecomment-552820322

> Instead of having security or licensing registering a global app filter, we can just recommend that each plugins register their own filter for their applications.

As I can see, the current implementation works differently: every updater is called with every app. Am I wrong or it's expected behavior?

Reply:
Currently yes, as that was the initial plan. Updating only a specific app can still easily be done with a simple id-check condition in the updater. However I agree that if we prefer per-app updater, this should probably change. Advantage of current implementation is that it allows to do both. But if we know that we only want per-app status updater, this can changes (propositions are in https://github.com/elastic/kibana/issues/45291#issuecomment-552820322).

Reply:
> However I agree that if we prefer per-app updater, this should probably change to forbid updating other plugin's apps.

I'd prefer this API

Reply:
I prefer it too, as long as we are sure about the decision to make this only per-app and nothing globally managed in some place. It also makes more sense in term of actual responsibility.

@joshdover are you good with this ?

Also @restrry, any preference in the API, between the in-app declaration and outside declaration ?

*in-app*
```typescript
const statusUpdaters$ = new BehaviorSubject<AppStatusUpdater>(() => ({}));

core.application.register({
  id: 'ml',
  statusUpdater$,
  mount( ... ) { },
})
``` 

*outside*
```typescript
core.application.register({
  id: 'myApp',
  mount( ... ) { },
})

const statusUpdaters$ = new BehaviorSubject<AppStatusUpdater>(() => ({}));

core.application.registerStatusUpdater('myApp', statusUpdaters$)
```

I *think* the first one makes more sense, as everything is declared at the same place, but WDYT ?

Reply:
👍 for `in-app` as providing high cohesion for the API